src/nodes/NodeConnectionManager.ts

```diff
@@ -1745,19 +1784,39 @@ class NodeConnectionManager {
    * Returns a promise that resolves once the connection has authenticated,
    * otherwise it rejects with the authentication failure
    * @param nodeId
+   * @param ctx
    */
-  public async isAuthenticatedP(nodeId: NodeId): Promise<void> {
+  public async isAuthenticatedP(
+    nodeId: NodeId,
+    ctx?: Partial<ContextTimedInput>,
+  ): Promise<void>;
+  @timedCancellable(true)
+  public async isAuthenticatedP(
+    nodeId: NodeId,
+    @context ctx: ContextTimed,
+  ): Promise<void> {
     const targetNodeIdString = nodeId.toString() as NodeIdString;
     const connectionsEntry = this.connections.get(targetNodeIdString);
     if (connectionsEntry == null) {
       throw new nodesErrors.ErrorNodeConnectionManagerConnectionNotFound();
     }
+    const { p: abortP, rejectP: triggerAbort } = utils.promise<void>();
+    const abortHandler = () => {
+      triggerAbort(ctx.signal.reason);
+    };
+    if (ctx.signal.aborted) {
+      triggerAbort(ctx.signal.reason);
+    } else {
+      ctx.signal.addEventListener('abort', abortHandler, { once: true });
+    }
     try {
-      return await connectionsEntry.authenticatedP;
+      return await Promise.race([connectionsEntry.authenticatedP, abortP]);
```

Comment:
Just a note, the reason `abortP` promise type should be `never` is so that this `Promise.race` can only return the `authenticatedP` type. It's *fine* here because both promises are void types. But if `authenticatedP` was any other type then the return type would complain here.

Basically the `never` type  means you can never pass this point unless it's an error. The `utils.never()` does this too. It throws an error and the return type is never because it never returns. This means that any code after it *should* be dead code in the eyes of the typescript compiler.